src/mpn_extras.h

```diff
@@ -574,6 +579,8 @@ FLINT_DLL extern const flint_mpn_mulhigh_normalised_func_t flint_mpn_mulhigh_nor
 # define FLINT_MPN_MULHIGH_FUNC_TAB_WIDTH 8
 # define FLINT_MPN_SQRHIGH_FUNC_TAB_WIDTH 8
 # define FLINT_MPN_MULHIGH_NORMALISED_FUNC_TAB_WIDTH 0
+# define FLINT_MPN_SQRHIGH_NORMALISED_FUNC_TAB_WIDTH 0
+
```

Comment:
Should be 2, right?

Reply:
No, the tab for `sqrhigh_normalised` is distinct from `sqrhigh`. The reason being that these hardcoded routines gain a lot from having the result still in registers. This doesn't matter as much as $n$ grow bigger, but at these sizes it does really matter.

Reply:
Having the result in registers before shifting (or adc-chain), that is.